### ISSUE TITLE
dropselect doesn't update index correctly

### ISSUE THREAD

**Issue by yagui** [ppizarror/pygame-menu#340]
**Environment information**
Describe your environment information, such as:
- SO: linux
- python version: v3.8
- pygame version: v2.0.1
- pygame-menu version: v4.0.6

**Describe the bug**
Updating a `dropselect` item list with fewer elements might raise an exception when the selected index is higher than new list lenght. In the following example the button deletes the item selected, but fails if it is the last item.

**To Reproduce**
```
import pygame, pygame_menu

SCREENWIDTH  = 800
SCREENHEIGHT = 480

def main():
    pygame.init()

    surface = pygame.display.set_mode((SCREENWIDTH, SCREENHEIGHT))

    def test_1(select):
        sVal = select.get_value()
        items = select._items
        items.pop(items.index(sVal[0]))
        #select._index = -1
        select.update_items(items)
        select.make_selection_drop()

    menu = pygame_menu.Menu('Test', SCREENWIDTH, SCREENHEIGHT,
                            theme = pygame_menu.themes.THEME_DARK,
                            onclose = pygame_menu.events.RESET,
                            touchscreen = False,
                            joystick_enabled = False,
                            mouse_enabled = True,
                            )

    S1 = menu.add.dropselect('Subject Id', items=[('a',),('b',),('c',),('d',),('e',),('f',)], dropselect_id='s0')
    menu.add.button('One',test_1, S1)

    # Allows menu to be run
    menu.mainloop(surface)
    print(menu.get_input_data())
    return

if __name__ == "__main__":
    main()
```

**Expected behavior**
The documentation says that if the new list has different size than the actual list, the new index will be -1, but that is not happening.
https://pygame-menu.readthedocs.io/en/4.0.6/_modules/pygame_menu/widgets/widget/dropselect.html#DropSelect.update_items
As a workaround, uncommenting the line (15) that sets `_index` to -1, helps to avoid the exception.

Also, I don't understand what `select.make_selection_drop()` but there is an exception if I don't call it.

**Additional context**
I belived this verification
https://github.com/ppizarror/pygame-menu/blob/3d8ef71b882a6e7bcbe7299af8d16fbcc0875d8f/pygame_menu/widgets/widget/dropselect.py#L1000
is intended to be before
https://github.com/ppizarror/pygame-menu/blob/3d8ef71b882a6e7bcbe7299af8d16fbcc0875d8f/pygame_menu/widgets/widget/dropselect.py#L992

**Comment by yagui** [comment]
I can see that instead of providing a new list, I'm deleting an element of the actual list. Is there another better way to remove an item from the droplist?

**Comment by ppizarror** [comment]
That is clearly an issue, it should work with any update list. Thanks for letting me know, dropselect was quite complicated to implement

**Comment by ppizarror** [comment]
Fixed in #341; there were two main issues.

First, the docs were not correct as it previously said "...of the select will be the first item of the list", but that was not the behavior of the menu, as the index was set to -1 instead.
Second, there was an issue regarding your example, as you accessed to private _items list without copying, thus, after modifying the list the length was the same, then, the index did not change. For such purpose, I've added a new method ``get_items`` that returns a copy of the list.
